lib/npg_pipeline/function/autoqc.pm

```diff
@@ -177,62 +183,80 @@ sub _create_definition_object {
-  }
+  my $archive_path = $self->archive_path;
+  my $recal_path= $self->recalibrated_path;
+  my $dp_archive_path = $dp->path($self->archive_path);
+  my $cache10k_path = $dp->short_files_cache_path($archive_path);
+  my $qc_out_path = $dp->qc_out_path($archive_path);
+  my $bamfile_path = File::Spec->catdir($dp_archive_path, $dp->file_name(ext => 'bam'));
+  my $fq1_filepath = File::Spec->catdir($cache10k_path, $dp->file_name(ext => 'fastq', suffix => '_1'));
+  my $fq2_filepath = File::Spec->catdir($cache10k_path, $dp->file_name(ext => 'fastq', suffix => '_2'));
+  my $fqt_filepath = File::Spec->catdir($cache10k_path, $dp->file_name(ext => 'fastq', suffix => '_t'));
+  my $fqc1_filepath = File::Spec->catdir($dp_archive_path, $dp->file_name(ext => 'fastqcheck', suffix => '1'));
+  my $fqc2_filepath = File::Spec->catdir($dp_archive_path, $dp->file_name(ext => 'fastqcheck', suffix => '2'));
+
+  my $c = sprintf '%s --check=%s --rpt_list="%s" --filename_root=%s --qc_out=%s',
+                  $QC_SCRIPT_NAME, $check, $dp->{rpt_list}, $dp->file_name_root, $qc_out_path;
 
   if ($check eq q[insert_size]) {
     $c .= $self->is_paired_read() ? q[ --is_paired_read] : q[ --no-is_paired_read];
   } elsif ($check eq q[qX_yield] && $self->platform_HiSeq) {
     $c .= q[ --platform_is_hiseq];
   }
 
-  my $qc_out = defined $tag_index ? $self->lane_qc_path($position) : $self->qc_path();
-  $qc_out or $self->logcroak('Failed to get qc_out directory');
-  my $qc_in  = defined $tag_index ? $self->lane_archive_path($position) : $self->archive_path();
+  ###########
+  # set qc_in
+  ###########
+  if(any { /$check/ } qw( gc_fraction qX_yield insert_size ref_match sequence_error)) {
+    $c .= qq{ --qc_in=$cache10k_path} 
+  }
+  elsif(any { /$check/ } qw( adapter genotype verify_bam_id upstream_tags )) {
+    $c .= qq{ --qc_in=$archive_path} 
+  }
+  else {
+    ## default qc_in??
+  }
+
+  #################
+  # set input_files
+  #################
   ##no critic (ControlStructures::ProhibitCascadingIfElse)
-  if ($check eq q[adapter]) {
-    $qc_in  = defined $tag_index
-              ? File::Spec->catfile($self->recalibrated_path(), q[lane] . $position)
-              : $self->recalibrated_path();
-  } elsif ($check eq q[spatial_filter] && $is_multiplexed_lane) {
-    $qc_in .= (q[/lane] . $position);
-  } elsif ($check eq q[tag_metrics]) {
-    $qc_in = $self->bam_basecall_path();
-  } elsif ($check eq q[sequence_error] or $check eq q[ref_match] or $check eq q[insert_size]) {
-    $qc_in .= q[/.npg_cache_10000]
+  if(any { /$check/ } qw( gc_fraction qX_yield)) {
+    $c .= qq[ --input_files=$fqc1_filepath --input_files=$fqc2_filepath]
+  }
+  elsif(any { /$check/ } qw( insert_size ref_match sequence_error )) {
+    $c .= qq[ --input_files=$fq1_filepath --input_files=$fq2_filepath];
+  }
+  elsif(any { /$check/ } qw( adapter genotype verify_bam_id )) {
+    $c .= qq{ --input_files=$bamfile_path} # note: single bam file 
+  }
```

Comment:
Adapter might have to be treated separately since it's running parallel to seq_alignment and uses unaligned bam files. Does this mean that it will run on unmerged files? This will break assumptions I made in qc_store. We need to discuss this.